x-pack/plugins/security_solution/public/detection_engine/rule_response_actions/osquery/osquery_response_action.tsx

```diff
@@ -85,8 +81,7 @@ export const OsqueryResponseAction = React.memo((props: OsqueryResponseActionPro
       );
     }
 
-    // @ts-expect-error ts upgrade v4.7.4
-    if (isMounted() && OsqueryForm) {
```

Comment:
@gergoabraham I believe we had some issues here (not typescript but how it works). Are you sure this change preserves the behavior?

Reply:
I am mostly sure, yes...

`OsqueryForm` comes from `useKibana().services.osquery.OsqueryResponseActionTypeForm`, which [got its value](https://github.com/gergoabraham/kibana/blob/chore/defend-workflows-fix-ts-4.7.4-type-errors/x-pack/plugins/osquery/public/plugin.ts#L138) during plugin start, and its value is the return value of `getLazyOsqueryResponseActionTypeForm()`, which - independently of the lazy load - [returns a function component](https://github.com/gergoabraham/kibana/blob/chore/defend-workflows-fix-ts-4.7.4-type-errors/x-pack/plugins/osquery/public/shared_components/lazy_osquery_action_params_form.tsx) which itself is never falsy, so based on this, I believe there should be no problem.

...but...

I'm happy to test it if you can help what should be tested here. I did some basic tests (i.e. tried to use this component, by throttling network connection so it loads slowly) and there were no issues.

Reply:
I remember there was some strange behaviour that is why I made this OqueryForm check. I'll test it manually to see if this is not broken now. I am saying this just because there is a strange behavior in the test. It's not succesful - it's pending... 
<img width="819" alt="Zrzut ekranu 2023-09-12 o 16 14 11" src="https://github.com/elastic/kibana/assets/16632552/a35d7870-2b2e-4b10-bdb1-03f5565dfc68">

Reply:
I tested this manually and I think it still works as expected, thanks! 
However - can you please do me a favor and remove the browser filter from here? Lets see if this make the test run then. https://github.com/elastic/kibana/blob/4180a1a105053e5579e6852082bdeaa06f019cd3/x-pack/plugins/osquery/cypress/e2e/all/alerts_response_actions_form.cy.ts#L33C16-L33C16

Reply:
thank you for testing it!

removed the browser filters, let's see what happens: dac76a59c595be885b2f4d8c3671ab15c0567c33

Reply:
I think something is broken, moreover the artifacts are not build. Feel free to revert this commit and just ship your changes. I'll take care of osquery testing in a separate PR. How does this sound?

Reply:
revert is done, thanks for the help, @tomsonpl!
cf43db45a5248121317eefada884154fc3e64bb0